packages/css/src/plugins/expand-shorthands/utils.tsx

```diff
@@ -0,0 +1,78 @@
+import { Node } from 'postcss-values-parser';
+
+/**
+ * Common global values
+ */
+export const globalValues = ['inherit', 'initial', 'unset'];
+
+/**
+ * Returns `true` if the node is a color,
+ * else `false`.
+ *
+ * @param node
+ */
+export const isColor = (node: Node) => {
+  if (node.type === 'word' && node.isColor) {
+    return true;
+  }
+
+  if (node.type === 'func' && node.isColor) {
+    return true;
+  }
+
+  return false;
+};
+
+/**
+ * Returns `true` if the node is a width,
+ * else `false`.
+ *
+ * @param node
+ */
+export const isWidth = (node: Node) => {
+  if (node.type === 'numeric') {
+    if (
+      [
+        'px',
+        'rem',
+        'em',
+        '%',
+        'pt',
+        'cm',
+        'mm',
+        'Q',
+        'in',
+        'pc',
+        'ex',
+        'ch',
+        'lh',
+        'vw',
+        'vh',
+        'vmin',
+        'vmax',
+        'fr',
+      ].includes(node.unit)
+    ) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+/**
+ * Returns calculated width of a node.
+ *
+ * @param node
+ */
+export const getWidth = (node: Node) => {
+  if (node.type === 'numeric') {
+    return `${node.value}${node.unit}`;
+  }
+
+  if (node.type === 'word') {
+    return node.value;
+  }
+
+  return undefined;
```

Comment:
Q: This should be `0` or `auto` or `undefined`? 🤔

Reply:
undefined or throw an error TBH 😄 for now ive just gone with undefined

what do you think?

Reply:
update: actually undefined is useful because we end up returning invalid (ie empty rules) if it returns undefined.